### PR TITLE
Add rate limiter to limit disk IO when doing compaction and flush

### PR DESCRIPTION
**Pull request by zhangjinpeng1987**
Please merge after this [pr](https://github.com/facebook/rocksdb/pull/1455) has merged by rocksdb.

### CODE REVIEW

File: src/rocksdb_options.rs [hunk 1]
```diff
@@ -542,6 +568,15 @@ impl Options {
             rocksdb_ffi::rocksdb_options_set_num_levels(self.inner, n);
         }
     }
+
+    pub fn set_ratelimiter(&mut self, rate_bytes_per_sec: i64) {
+        let rate_limiter = RateLimiter::new(rate_bytes_per_sec,
+                                            100 * 1000 /* 100ms should work for most cases */,
+                                            10 /* should be good by leaving it at default 10 */);
+        unsafe {
+            rocksdb_ffi::rocksdb_options_set_ratelimiter(self.inner, rate_limiter.inner);
```
Comment (siddontang) [inline src/rocksdb_options.rs]: seem that we will call RateLimiter drop directly after leave set_ratelimiter. Is it safe?

Comment (zhangjinpeng1987) [inline src/rocksdb_options.rs]: Yes

Comment (siddontang) [inline src/rocksdb_options.rs]: This will call rocksdb_ratelimiter_destroy directly...

Comment (zhangjinpeng1987) [inline src/rocksdb_options.rs]: It is safe, `rocksdb_options_set_ratelimiter` will take the inner ratelimiter away, in that case the `rocksdb_ratelimiter_destory` just free the wrapper.

**Comment by siddontang** [comment]
LGTM

**Comment by zhangjinpeng1987** [comment]
@BusyJay PTAL

**Comment by BusyJay** [comment]
LGTM

**Pull request merged**